### PR TITLE
fix: broken and redirected links in docs homepage

### PR DESCRIPTION
**Pull request by manika-signoz**
## Pull Request

---

---

### ✅ Change Type
_Select all that apply_

- [ ] ✨ Feature
- [x] 🐛 Bug fix
- [ ] ♻️ Refactor
- [ ] 🛠️ Infra / Tooling
- [ ] 🧪 Test-only

---

---

- Tests added/updated: NA
- Manual verification: Done on preview.
- Edge cases covered: None in current PR.

---

- Blast radius: Docs homepage links
- Potential regressions: None
- Rollback plan: Not needed, if any issues, we should fix those but PR has low blast radius due to url only changes.

---

### 📋 Checklist
- [x] Tests added or explicitly not required
- [x] Manually tested
- [x] Breaking changes documented
- [x] Backward compatibility considered

---

## 👀 Notes for Reviewers

In a follow up, fixing the workflows and url checks.

---

### CODE REVIEW

**Review by crazyuploader — APPROVED** [review]

File: app/(site)/docs/introduction/page.tsx [hunk 1]
```diff
@@ -33,7 +33,7 @@ export default async function DocsIntroductionPage() {
         title="Explore the rest of SigNoz"
         description="Once your data is flowing in — go deeper into what we offer."
         guidesCount={12}
-        viewAllHref="/docs/userguide/"
+        viewAllHref="/docs/querying/overview/"
```
Comment (makeavish) [inline app/(site)/docs/introduction/page.tsx:36]: Why querying doc here?

**Review by makeavish — APPROVED** [review]
